rfcs/0024-incremental-ledger-catchup.md

```diff
@@ -0,0 +1,81 @@
+## Terminology
+
+history root: the oldest node in root history
+
+## Summary
+
+The origin transition catchup design has performance issues. In
+current implementation, we would always downloading a list of transitions
+from the requested hash to peer's history root.
+The minimum number of transitions that transition catchup would request is
+k + 1 under current implementation. This is not very efficient in a realistic
+setting where k is ~3000. The new design features
+1) avoids requesting unnecessary transitions;
+2) instead of having a monolithic request to download everything; try to
+   download small pieces from different peers at the same time.
+
+## Motivation
+
+The new design would improve the efficiency of doing ledger catchup.
+
+## Detailed design
+
+Split ledger catchup into 2 phases:
+
+1) Instead of
+   * requesting a path/list of transitions from peer's history root to the requested hash,
+   * requesting a merkle path/list from peer's history root to the requested hash together with their root history transition.
+   
+   * The merkle path/list contains a list of *state_body_hash*es as its
```

Comment:
I like this solution. We can download the list of state body hashes and use them to construct the state_hash. We can then in parallel download the external_transitions in and asynchronously build the external_transitions into breadcrumbs.